gensim/models/doc2vec.py

```diff
@@ -585,18 +592,22 @@ def _vocab_from(self, documents, progress_per=10000):
                 interval_start = default_timer()
                 interval_count = total_words
             document_length = len(document.words)
+
             for tag in document.tags:
                 self.docvecs.note_doctag(tag, document_no, document_length)
+
             for word in document.words:
-                total_words += 1
-                if word in vocab:
-                    vocab[word].count += 1
-                else:
-                    vocab[word] = Vocab(count=1)
+                vocab[word] += 1
+
+            if self.max_vocab_size and len(vocab) > self.max_vocab_size:
+                total_words += utils.prune_vocab(vocab, min_reduce)
+                min_reduce += 1
```

Comment:
Agreed that it will be somewhat peculiar for the size of `raw_vocab` to be a blocking issue: except in weird cases, it's smaller than what will come later. And, exactly the corner cases where pruning might help, the user will have to apply more thought – and might want to graduate to extra corpus pre-processing as a distinct step, or some other more-efficient/probabilistic structure.

Farther out, I wonder if something like the count-min-sketch code (possibly still with luxurious amounts of RAM) could enable simultaneous vocabulary and bigram/trigram survey. That might argue for moving the word-survey out of W2V/D2V, where it might be a really simple implementation or very-sophisticated, and W2V/D2V just works with the results. (Have to think about that some more...)